packages/test/test-end-to-end-tests/src/test/detachedContainerTests.spec.ts

```diff
@@ -898,4 +898,51 @@ describeNoCompat("Detached Container", (getTestObjectProvider) => {
 			"Stored handle should match!!",
 		);
 	});
+
+	it("Compressors sync after detached container attaches and sends an op", async () => {
+		provider.reset();
+		const testConfig = {
+			...testContainerConfig,
+			runtimeOptions: {
+				...testContainerConfig.runtimeOptions,
+				enableRuntimeIdCompressor: true,
+			},
+		};
+		loader = provider.makeTestLoader(testConfig) as Loader;
+		const container = await loader.createDetachedContainer(provider.defaultCodeDetails);
+
+		// Get the root dataStore from the detached container.
+		const response = await container.request({ url: "/" });
+		const dataStore = response.value as ITestFluidObject;
+		const testChannel1 = await dataStore.getSharedObject<SharedCell>(sharedCellId);
+
+		// Generate an Id before attaching the container
+		(testChannel1 as any).runtime.idCompressor.generateCompressedId();
+		// Attach the container. The generated Id won't be synced until another op
+		// is sent after attaching becuase most DDSs don't send ops until they are attached.
+		await container.attach(request);
+
+		// Create another container to test sync
+		const url: any = await container.getAbsoluteUrl("");
+		const loader2 = provider.makeTestLoader(testConfig) as Loader;
```

Comment:
nit: You don't need to create another loader. You can do `const container2 = await provider.loadTestContainer(testConfig)`